nixos/lib/testing/interactive.nix

```diff
@@ -1,6 +1,97 @@
 { config, lib, moduleType, hostPkgs, ... }:
 let
-  inherit (lib) mkOption types mdDoc;
+  inherit (lib) mkOption mkDefault types mdDoc concatStringsSep mapAttrsToList;
+  inherit (builtins) toString attrNames;
+  finalNodes = config.interactive.driver.nodes;
+
+  # allow accessing machines through a jumphost whose ssh port is forwarded to from the host
+  sshConfig = hostPkgs.writeText "ssh-config" ''
+    Host *
+      User root
+      StrictHostKeyChecking no
+      BatchMode yes
+      ConnectTimeout 20
+      UserKnownHostsFile=/dev/null
+      LogLevel Error # no "added to known hosts"
+    Host redeploy_jumphost
+      HostName localhost
+      Port ${
+        let
+          fwds = config.interactive.nodes.redeploy_jumphost.virtualisation.forwardPorts;
+          sshFwds = builtins.filter (fwd: fwd.guest.port == 22) fwds;
+          sshFwd = builtins.head sshFwds;
+        in
+          toString sshFwd.host.port
+        }
+    Host * !redeploy_jumphost
+      ProxyJump redeploy_jumphost
+  '';
+
+  redeploy = hostPkgs.writeShellScriptBin "redeploy" ''
```

Comment:
I *think* I'd prefer for this to be implemented in the (python) driver instead, but I'll defer to @tfc.

Reply:
I'd love if `redeploy` could be put in the python driver. It could maybe be more powerful in that case, since it could also deploy new machines / power off ones that have been decommissioned.

But I'm not sure how that's possible. The python driver only has access to the configuration that existed at the time it was spun up. The whole point is that we then reconfigure the machines, and while the old driver is still running, redeploy the new configurations.